govcd/edgegateway.go

```diff
@@ -1076,3 +1076,88 @@ func (eGW *EdgeGateway) buildProxiedEdgeEndpointURL(optionalSuffix string) (stri
 
 	return hostname, nil
 }
+
+// GetLoadBalancer retrieves load balancer configuration of `&types.LoadBalancer` and can be used
+// to access global configuration options. These are 4 fields only:
+// LoadBalancer.Enabled, LoadBalancer.AccelerationEnabled, LoadBalancer.Logging.Enable,
+// LoadBalancer.Logging.LogLevel
+func (egw *EdgeGateway) GetLoadBalancer() (*types.LoadBalancer, error) {
+	if !egw.HasAdvancedNetworking() {
+		return nil, fmt.Errorf("only advanced edge gateway supports load balancing")
+	}
+
+	httpPath, err := egw.buildProxiedEdgeEndpointURL(types.LBConfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("could not get Edge Gateway API endpoint: %s", err)
+	}
+
+	loadBalancerConfig := &types.LoadBalancer{}
+	_, err = egw.client.ExecuteRequest(httpPath, http.MethodGet, types.AnyXMLMime,
+		"unable to read load balancer configuration: %s", nil, loadBalancerConfig)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return loadBalancerConfig, nil
+}
+
+// UpdateLoadBalancerGlobal allows to update global load balancer configuration.
+// These are 4 fields only: LoadBalancer.Enabled, LoadBalancer.AccelerationEnabled,
+// LoadBalancer.Logging.Enable, LoadBalancer.Logging.LogLevel. They are represented in load balancer
+// global configuration tab in the UI.
+// All other fields are ignored and sent as they are in order to prevent load balancer configuration
+// corruption
+func (egw *EdgeGateway) UpdateLoadBalancerGlobal(lb *types.LoadBalancer) (*types.LoadBalancer, error) {
+	if !egw.HasAdvancedNetworking() {
+		return nil, fmt.Errorf("only advanced edge gateway supports load balancing")
+	}
+
+	if err := validateUpdateLoadBalancerGlobal(lb); err != nil {
+		return nil, err
+	}
+	// Retrieve load balancer to work on latest configuration
+	currentLb, err := egw.GetLoadBalancer()
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve load balancer before update: %s", err)
+	}
+
+	// Modify only the global configuration settings
+	currentLb.Enabled = lb.Enabled
+	currentLb.AccelerationEnabled = lb.AccelerationEnabled
+	currentLb.Logging = lb.Logging
+	// Ommit the version as it is updated automatically with each put
+	currentLb.Version = ""
+
+	// Push updated configuration
```

Comment:
Should we have a check here to see whether the intended fields are already set as we want them?
If so, we could just return the current LB without further action

Reply:
Very good spot. Done.